sn_networking/src/record_store.rs

```diff
@@ -79,6 +79,8 @@ pub struct NodeRecordStore {
     encryption_details: (Aes256GcmSiv, [u8; 4]),
     /// Time that this record_store got started
     timestamp: SystemTime,
+    /// List of records that got obsoleted, i.e. overwritten by a newer version
+    obsoleted_records: HashSet<(NetworkAddress, RecordType)>,
```

Comment:
Hmmm, I think this way memleaks lie... 

This is essentially about register versions for us (though I wonder if spends might suffer from similar issues?)


A potential strategy here is merging registers if we're getting different content in... But that feels like it should be left up to the client... 

So I find myself wondering what we actually want to be doing with registers here. We have different versions on different nodes. Presumably the client only updated one node (is that a bug for edits?)

Either way, we can end up in this situation through conn issues. 

Should nodes not just keep the verison they have until they receive an actual PUT? 

And if we do that, is all we need to be doing here NOT requesting updates if we have the key/type? (ie `Register`, ignoring the content hash), and NOT judge nodes as bad if they have differing content? (As that's a client world issue?)

Reply:
> all we need to be doing here NOT requesting updates if we have the key/type?

If doing so, it means nodes will miss the newer version of the register.
As the new version will only be sent to one payee node (due to the store_cost to pay), and then be replicated to others from there.

Same reason for the following, that we cann't do that, as a client direct PUT is not to ALL nodes.

> Should nodes not just keep the verison they have until they receive an actual PUT?